In [1]:
import MeCab
import unidic
print(unidic.DICDIR)
MeCab.Tagger("-d" + unidic.DICDIR)

/home/yamada/.local/lib/python3.6/site-packages/unidic/dicdir


<Swig Object of type 'MeCab::Tagger *' at 0x7f82976fceb8>

- よくわからないもの

    
    

In [2]:
import json
import os
import pandas as pd
import numpy as np
from pathlib import Path
import collections
from sklearn.model_selection import train_test_split
from sklearn import metrics

import sys
sys.path.append("../")
from datatools.analyzer import *
from utterance.error_tools import *

from datatools.maneger import DataManager
from datatools.preproc import Preprocessor

In [3]:
ngram_path = "../../corpus/google/3-gram/"

In [4]:
key1_set = set( os.listdir(ngram_path) ) 

In [17]:
path = "../eval_labeled/"
datalist = ['DCM', 'DIT', 'IRS']
convs = read_conv(path, datalist)

error = "Grammatical error"
errors = ['Grammatical error', "Uninterpretable"]
# errors = ['Semantic error', "Wrong information"]
sys_utt = []
y = []
for conv in convs:
    for ut in conv:
        if ut.is_system() and ut.is_exist_error():
            if not ut.utt[-1] in ["？", "！", "。", "!"]:
                sys_utt.append( clean_text( ut.utt+"。" ))
                # sys_utt.append(ut.utt)
            else:   
                sys_utt.append(ut.utt)
            if ut.is_error_included(errors):
                y.append(1)
            else:
                y.append(0)

In [6]:
print("len of y:{0}, error '{1}' counts:{2}".format(len(y), errors, y.count(1)))

len of y:1386, error '['Semantic error', 'Wrong information']' counts:403


In [7]:
def load_dict(path, key1):
    filename = key_formated(key1)
    with open(path+filename, "r") as f:
        key1_dict = json.load(f) 
    return key1_dict  

In [8]:
def key_formated(key1):
    return "ngram_"+key1+".json"

In [9]:
key2dict = dict()
registerform_dict = dict()
analyzer_unidic=  Analyzer(Tokenizer())

In [10]:
#  を考慮
# 何故か↑が出てくる
def judge_key3(key1_dict, ngram):
    key1 = ngram[0]
    key2 = ngram[1]
    key3 = ngram[2]

    if key2 not in key1_dict[key1]:
        # '"'をkeyにして聞いてみる
        if '"' in key1_dict[key1] and key2 in key1_dict[key1]['"']:
            return True
        if '、' in key1_dict[key1] and key2 in key1_dict[key1]['、']:
            return True
        return False
    
    if key3 not in key1_dict[key1][key2]:
        return False

    return True 

In [11]:

def search_registerforam(key, reading):
    filename = key_formated(key)
    if filename in key1_set:
        return key
    # 登録されていない
    else:
        for other in reading:
            filename = key_formated(key)
            if filename in key1_set:
                registerform_dict[key] = other
                return other
        return key


def sentence_judge(sentence, N=3):
    
    df = analyzer_unidic.analyze_with_dataframe(sentence)
    
    # 登録
    sentence_ = ""
    for txt, read in zip(df.surface, df.reading):
        sentence_ += search_registerforam(txt, read)
    sents = sentence2docs(sentence_, sents_span=True)
    # print(sents)

    # print(tokens)
    ngram_text = []
    ngram_pos = []

    for sent in sents:
        # 各文章 : sent
        df = analyzer_unidic.analyze_with_dataframe("".join([token.orth_ for token in sent]))
        surfaces = []
        poses = []
        readings = []
        for txt, p, read in zip(df.surface, df.part_of_speech, df.reading):
            surfaces.append(txt)
            readings.append(read)
            poses.append(p)

            for i in range(len(surfaces)-N+1):
                # print(L[i:i+N])
                ngram_text.append(surfaces[i:i+N])
                ngram_pos.append(poses[i:i+N])
                # ngram_reading.append(readings[i:i+N])
    
    # print(ngram_text)
    # return
    
    for ngram, pgram in zip(ngram_text, ngram_pos):
        # if "助動詞" in pgram[1] or "助詞" in pgram[1] or "助動詞" in pgram[0] or "助詞" in pgram[0]:
        # if "助動詞" in pgram[0] or "助詞" in pgram[0]:
        if "助詞" in pgram[1] and "でしょう" != ngram[2] :
            # print("\tcontext : {0}| ->".format(ngram))
            
            key1 = ngram[0]
            if key_formated(key1) not in key1_set:
                continue

            if key1 in key2dict:
                key1_dict = key2dict[key1]
            else:
                key1_dict = load_dict(ngram_path, key1)
                key2dict[key1] = key1_dict

            # チェック
            if not judge_key3(key1_dict, ngram):
                print(ngram)
                # reading を調整
                
                # 見つかれば，その語彙を登録
                return False
    return True
            


In [12]:
y_pred_2 = []
for i, utt in enumerate( tqdm( sys_utt ) )  :
    cleaned = clean_text(utt).replace("?", "。").replace("!", "")
    # print(cleaned)
    if sentence_judge(cleaned, N=3):
        y_pred_2.append(0)
    else:
        y_pred_2.append(1)
    # if i==35:
    #     break

  0%|          | 1/1386 [00:00<13:17,  1.74it/s]

['遍路', 'の', '逆回り']


  4%|▍         | 55/1386 [00:18<02:53,  7.68it/s] 

['天麩羅', 'は', '食べる']
['寿司', 'は', '縁側']
['なん', 'と', '。']


  5%|▌         | 70/1386 [00:23<03:40,  5.98it/s]

['渾名', 'は', 'あり']
['渾名', 'は', 'あり']
['渾名', 'を', '持ち']
['同僚', 'を', '渾名']
['同僚', 'を', '渾名']


  5%|▌         | 74/1386 [00:23<02:28,  8.84it/s]

['掛け声', 'が', '出す']
['掛け声', 'を', '覚える']
['大声', 'で', '上げる']


  6%|▌         | 82/1386 [00:24<01:24, 15.38it/s]

['映画', 'は', '一人']


  6%|▋         | 89/1386 [00:24<01:15, 17.29it/s]

['位', 'ヶ', '岳']
['位', 'ヶ', '岳']
['デート', 'で', '御洒落']


  7%|▋         | 96/1386 [00:24<01:19, 16.13it/s]

['勉強', 'も', '頑張ろう']
['フラワー', 'は', '手芸']


  7%|▋         | 103/1386 [00:25<01:09, 18.37it/s]

['外食', 'は', 'い']
['独身', 'で', '0']
['由紀恵', 'は', '怖い']


  8%|▊         | 109/1386 [00:25<01:03, 20.22it/s]

['デート', 'で', '御洒落']


  8%|▊         | 112/1386 [00:29<08:23,  2.53it/s]

['争い', 'が', '起ころう']


  8%|▊         | 114/1386 [00:31<11:58,  1.77it/s]

['と', 'か', '泣き']


  9%|▊         | 118/1386 [00:32<07:49,  2.70it/s]

['なら', 'を', '使う']
['池', 'は', '寒い']


 10%|▉         | 136/1386 [00:33<01:53, 11.00it/s]

['微熱', 'は', '多い']


 11%|█         | 149/1386 [00:35<02:08,  9.60it/s]

['汗', 'は', 'かっこ']


 11%|█▏        | 159/1386 [00:37<02:35,  7.88it/s]

['さん', 'の', '朝御飯']
['宇宙', 'で', 'プレー']
['パー', 'より', '0']


 12%|█▏        | 171/1386 [00:37<01:16, 15.79it/s]

['時期', 'から', '資格']
['最近', 'じゃあ', '漫画']


 13%|█▎        | 174/1386 [00:37<01:18, 15.44it/s]

['干支', 'は', '辰年']
['です', 'か', '…']


 13%|█▎        | 183/1386 [00:38<01:13, 16.38it/s]

['本当', 'に', 'ラーメン']
['補給', 'は', '少ない']


 14%|█▍        | 192/1386 [00:38<00:50, 23.66it/s]

['料理', 'は', '手早い']


 14%|█▍        | 198/1386 [00:39<00:57, 20.53it/s]

['つける', 'と', 'ホッと']


 15%|█▌        | 209/1386 [00:39<00:42, 28.00it/s]

['料理', 'は', '手早い']
['独り暮らし', 'は', '食']


 16%|█▌        | 217/1386 [00:40<00:50, 23.07it/s]

['手', 'を', '貯金']
['手', 'を', '貯金']


 16%|█▌        | 224/1386 [00:40<00:54, 21.15it/s]

['頬', 'に', 'ピアス']
['耳朶', 'の', '大きい']
['金', 'は', '大き']


 16%|█▋        | 227/1386 [00:40<01:13, 15.78it/s]

['お', 'や', '、']


 17%|█▋        | 234/1386 [00:42<02:43,  7.03it/s]

['スイカ', 'は', '面白い']


 19%|█▊        | 257/1386 [00:43<00:47, 23.71it/s]

['料理', 'は', 'そろい']
['一人', 'で', '食べる']


 20%|█▉        | 276/1386 [00:44<00:31, 35.14it/s]

['ウェブ', 'と', 'ホーム']
['梅雨入り', 'は', 'いい']
['カタ', 'を', '使い']


 21%|██        | 291/1386 [00:44<00:28, 39.09it/s]

['料理', 'は', '手早い']
['泳ぐ', 'を', '覚える']
['泳ぐ', 'を', '繰り返す']
['泳ぐ', 'を', '覚え']
['泳ぐ', 'を', '覚え']
['泳ぐ', 'を', '覚える']


 22%|██▏       | 301/1386 [00:44<00:27, 39.30it/s]

['バニラ', 'は', '相性']
['た', 'を', '仕入れる']


 22%|██▏       | 311/1386 [00:44<00:29, 36.90it/s]

['た', 'は', '鮮度']


 23%|██▎       | 323/1386 [00:45<00:37, 28.21it/s]

['嗽', 'で', '予防']
['嗽', 'で', '予防']
['心掛ける', 'か', 'も']


 24%|██▍       | 332/1386 [00:46<00:42, 24.68it/s]

['天辺', 'を', '軽く']
['天辺', 'を', '軽く']
['ID', 'に', '行き']
['ID', 'に', '行き']


 24%|██▍       | 336/1386 [00:46<00:41, 25.03it/s]

['求める', 'か', 'も']
['せき', 'が', '欲しい']
['恋人', 'は', '重い']


 25%|██▍       | 343/1386 [00:46<00:47, 21.84it/s]

['本当', 'に', 'おもしろい']


 25%|██▌       | 353/1386 [00:47<00:47, 21.81it/s]

['天気', 'は', '恋しい']


 26%|██▌       | 363/1386 [00:47<00:44, 23.09it/s]

['教える', 'か', 'も']


 27%|██▋       | 372/1386 [00:48<00:55, 18.18it/s]

['摘み', 'は', 'おいしい']


 27%|██▋       | 378/1386 [00:48<00:52, 19.24it/s]

['感', 'って', 'すっきり']
['感', 'って', 'すっきり']
['デート', 'で', '御洒落']


 28%|██▊       | 387/1386 [00:48<00:53, 18.51it/s]

['アクセス', 'の', 'ステー']
['アクセス', 'が', '都心']
['日本', 'と', 'JR']


 28%|██▊       | 393/1386 [00:49<00:50, 19.84it/s]

['楽しむ', 'と', '花']


 29%|██▉       | 399/1386 [00:49<00:55, 17.91it/s]

['ハーバー', 'の', '0']
['なっ', 'しー', 'が']


 29%|██▉       | 401/1386 [00:49<01:22, 11.93it/s]

['商品', 'に', '今日']
['テレビ', 'は', 'サイド']


 29%|██▉       | 408/1386 [00:50<01:35, 10.21it/s]

['ジブリ', 'の', '踊る']
['ジブリ', 'の', '隣り']
['のぶ代', 'と', 'ハビエル']
['男', 'を', '立体']


 30%|██▉       | 410/1386 [00:51<01:35, 10.25it/s]

['つとめ', 'て', '今年']
['今', 'の', 'フォーブス']


 30%|██▉       | 412/1386 [00:51<01:40,  9.66it/s]

['熱', 'の', '緊急']
['アメリカ', 'で', 'ジャイアン']


 30%|███       | 416/1386 [00:51<01:59,  8.09it/s]

['先行き', 'に', '対する']
['IQ', 'と', 'いえ']
['映画', 'の', 'You']
['鎌倉', 'の', '全国']
['鎌倉', 'の', '笑点']


 30%|███       | 419/1386 [00:52<01:27, 11.09it/s]

['鎌倉', 'の', 'ファミリー']
['マート', 'が', '配送']
['家', 'に', '中井']
['鎌倉', 'の', '馬鹿']


 30%|███       | 422/1386 [00:52<01:19, 12.07it/s]

['サイト', 'の', 'ダビンチ']
['サイト', 'の', 'あま']


 31%|███       | 433/1386 [00:54<02:05,  7.59it/s]

['饂飩', 'の', '御']
['饂飩', 'の', '御']
['金団', 'で', 'ソフト']
['バンク', 'の', '大人']


 32%|███▏      | 438/1386 [00:54<01:22, 11.54it/s]

['ロボット', 'の', 's']
['バンク', 'の', '家庭']
['バンク', 'の', '家庭']
['バンク', 'の', '家庭']
['空輸', 'の', '話題']


 32%|███▏      | 440/1386 [00:55<01:39,  9.55it/s]

['山', 'の', '海遊']
['海月', 'が', '大好き']
['館内', 'の', '太陽']


 32%|███▏      | 446/1386 [00:55<01:10, 13.28it/s]

['ホワイト', 'と', 'カフェ']
['パッケージ', 'も', '御洒落']
['大阪', 'の', 'チョコレートラテポッキーブレンド']
['ゲゲゲ', 'の', '女房']
['ゲゲゲ', 'の', '女房']


 32%|███▏      | 450/1386 [00:55<01:06, 14.07it/s]

['保険', 'を', '観戦']
['連盟', 'の', 'ギャグ']
['菓子', 'の', '鶴']


 33%|███▎      | 452/1386 [00:56<01:13, 12.70it/s]

['旅', 'の', '指差し']
['味', 'の', 'shall']


 33%|███▎      | 457/1386 [00:56<01:00, 15.40it/s]

['蜜柑', 'に', '比べ']
['来年', 'に', 'プロ']
['用', 'に', '饂飩']
['ソックス', 'の', 'キャンデー']


 33%|███▎      | 459/1386 [00:56<01:26, 10.74it/s]

['ソックス', 'の', 'キャンデー']
['野球', 'の', 'Microsoft']
['Microsoft', 'の', '里田']


 33%|███▎      | 461/1386 [00:56<01:29, 10.36it/s]

['投手', 'が', '切っ掛け']
['ウェンリー', 'と', 'いう']
['だ', 'から', '荒野']


 33%|███▎      | 464/1386 [00:57<01:19, 11.54it/s]

['料理', 'を', '受賞']
['街', 'に', 'NHK']


 34%|███▍      | 471/1386 [00:57<01:03, 14.32it/s]

['NHK', 'の', '町歩き']
['レボリューション', 'は', 'NHK']
['協会', 'など', '0']


 34%|███▍      | 473/1386 [00:57<01:07, 13.44it/s]

['デュボア', 'の', '授賞']
['グランプリ', 'の', 'レプリカ']
['ゲゲゲ', 'の', '女房']
['ゲゲゲ', 'の', '女房']


 34%|███▍      | 476/1386 [01:02<08:51,  1.71it/s]

['は', 'は', 'ハードル']


 35%|███▍      | 481/1386 [01:02<04:42,  3.20it/s]

['NHK', 'の', '町歩き']
['レント', 'の', '永作']
['保険', 'が', '連載']


 35%|███▍      | 485/1386 [01:03<03:25,  4.39it/s]

['イングレス', 'の', 'レプリカ']
['イングレス', 'と', 'コンビニエンス']
['マクドナルド', 'の', '遠藤']


 35%|███▌      | 490/1386 [01:03<02:08,  6.97it/s]

['問題', 'で', 'マクドナルド']
['マクドナルド', 'で', 'リング']
['マクドナルド', 'の', 'リング']
['公園', 'と', '遠刈田']
['NHK', 'の', 'ローリー']


 36%|███▌      | 496/1386 [01:04<01:43,  8.57it/s]

['ゲゲゲ', 'の', '女房']
['NHK', 'の', '町歩き']
['実際', 'は', 'タモリ']


 36%|███▌      | 499/1386 [01:04<01:20, 11.08it/s]

['ランド', 'や', '東京']
['バレーボール', 'の', '八木']
['ディズニーシー', 'の', '職人']
['なっ', 'しー', 'の']
['チュウ', 'も', '使用']


 36%|███▋      | 505/1386 [01:04<01:01, 14.30it/s]

['最近', 'は', 'ヨーク']
['ヒルズ', 'と', 'JR']
['ヒルズ', 'から', 'さほど']
['ヒルズ', 'で', '000000000']
['アナウンサー', 'を', '御']


 37%|███▋      | 508/1386 [01:04<00:54, 16.09it/s]

['ダッシュ', 'など', '多く']
['テレビ', 'の', '林家']


 37%|███▋      | 512/1386 [01:05<01:40,  8.69it/s]

['流れ', 'を', '関西']
['NHK', 'の', 'ケーキ']


 37%|███▋      | 519/1386 [01:06<01:46,  8.12it/s]

['女子', 'の', 'ブーム']
['だ', 'から', '荒野']
['あっ', 'で', 'セアカゴケグモ']
['レント', 'の', 'タモリ']


 38%|███▊      | 525/1386 [01:06<01:04, 13.28it/s]

['小説', 'の', 'ブラタモリ']
['弦', 'は', '奈良']
['奈良', 'の', '閻']
['元気', 'や', 'を']
['タイガー', 'の', 'や']
['タイガー', 'の', 'デギュスタシオン']


 38%|███▊      | 528/1386 [01:06<00:56, 15.17it/s]

['スケート', 'の', '東京']
['恵比寿', 'の', '画家']
['年始', 'に', '泊まり']
['野薔薇', 'か', 'も']
['野薔薇', 'か', 'も']


 38%|███▊      | 533/1386 [01:07<01:08, 12.45it/s]

['野薔薇', 'か', 'もん']
['シリーズ', 'に', 'カレー']
['味', 'も', 'シーフード']


 39%|███▉      | 539/1386 [01:07<00:54, 15.50it/s]

['味', 'の', 'アムゼル']
['で', 'は', '日清']
['シリーズ', 'の', '濃厚']
['戦', 'と', 'ホンジュラス']


 39%|███▉      | 542/1386 [01:07<00:47, 17.69it/s]

['プレー', 'が', 'シドニー']
['パリーグ', 'に', '入団']
['球団', 'の', 'パリーグ']
['投手', 'の', 'シドニー']
['球団', 'の', 'パシフィック']


 40%|███▉      | 550/1386 [01:08<00:56, 14.80it/s]

['シドニー', 'の', 'ベンチャー']
['シドニー', 'の', '慣習']
['コロッケ', 'が', '自動']
['陽平', 'や', 'アルベルト']
['オランダ', 'で', '展示']


 40%|███▉      | 554/1386 [01:08<00:51, 16.22it/s]

['代表', 'の', 'アムステルダム']
['ポスト', 'の', 'joy']
['車', 'と', '有名']
['0', 'の', '木村']


 40%|████      | 558/1386 [01:09<01:08, 12.09it/s]

['クッキー', 'を', '踊り']
['明子', 'と', 'ローリー']


 40%|████      | 560/1386 [01:09<01:03, 13.02it/s]

['薯', 'の', '御薩']
['ポテト', 'の', 'ちょい']


 41%|████      | 566/1386 [01:09<01:06, 12.31it/s]

['スイーツ', 'は', 'どの']
['0', 'の', 'ドラマ']


 41%|████      | 568/1386 [01:09<01:09, 11.82it/s]

['し', 'て', '000000000']
['階', 'の', '劇団']


 41%|████▏     | 573/1386 [01:10<01:09, 11.70it/s]

['ミュージカル', 'は', '特別']
['劇場', 'に', 'インスパイアー']
['四季', 'の', '岩田']
['四季', 'の', '朝日']
['劇場', 'と', '旭山']


 42%|████▏     | 579/1386 [01:10<00:49, 16.46it/s]

['園', 'の', 'オーレ']
['海豹', 'が', '好き']
['カルビー', 'の', '海豹']
['漁師', 'の', '力め']
['園', 'の', 'チョコレート']
['園', 'も', '大体']


 42%|████▏     | 583/1386 [01:10<00:47, 16.77it/s]

['県', 'が', '桜ん坊']
['スシロー', 'は', '山形']
['ソックス', 'は', '鶴岡']


 42%|████▏     | 587/1386 [01:11<00:50, 15.89it/s]

['内', 'で', 'ツキノワ']
['暮らし', 'の', 'e']
['チュウ', 'で', '送料']
['グランプリ', 'で', '送料']


 43%|████▎     | 592/1386 [01:11<00:46, 16.92it/s]

['公園', 'と', 'ナゴヤ']
['グリコ', 'の', '名古屋']
['グリコ', 'の', '安納']
['グリコ', 'の', '安納']
['ストップ', 'の', 'ファミマ']


 43%|████▎     | 597/1386 [01:11<00:42, 18.40it/s]

['ストップ', 'の', '筋肉']
['ストップ', 'と', 'パステル']
['クリーム', 'で', '思い出し']
['流', 'の', '羽生']


 43%|████▎     | 601/1386 [01:12<00:44, 17.45it/s]

['戦', 'と', 'ホンジュラス']
['MVP', 'は', '前']
['さん', 'が', '二人']


 44%|████▎     | 606/1386 [01:12<01:00, 12.83it/s]

['投手', 'へ', 'メジャー']
['投手', 'は', 'キリン']
['キリン', 'の', 'サンタ']
['クロース', 'の', '御']
['映画', 'の', 'ミスター']


 44%|████▍     | 609/1386 [01:12<00:59, 13.02it/s]

['パリーグ', 'の', 'ミスター']
['球場', 'まで', '駆け付け']


 44%|████▍     | 613/1386 [01:13<01:18,  9.85it/s]

['東京', 'で', '鶴']
['ドール', 'に', '代表']
['大阪', 'の', '馬事']


 45%|████▍     | 618/1386 [01:13<00:54, 14.17it/s]

['マニア', 'は', '東京']
['春休み', 'で', 'とても']
['連合', 'の', 'キャラクター']
['連合', 'の', 'ギャグ']
['シーク', 'に', 'ノミネート']


 45%|████▌     | 624/1386 [01:13<00:45, 16.77it/s]

['歌手', 'の', '明石家']
['芸人', 'の', '大竹']
['あなた', 'が', 'リオネル']
['マリノス', 'の', 'リオネル']
['泰幸', 'も', '現在']


 45%|████▌     | 628/1386 [01:14<00:51, 14.66it/s]

['マリノス', 'の', '今野']
['マリノス', 'で', 'キャプテンシー']
['ヤマザキ', 'は', '天皇']


 46%|████▌     | 634/1386 [01:14<00:39, 18.92it/s]

['静岡', 'の', 'じゃがりこ']
['lush', 'は', '、']
['lush', 'と', 'は']
['宇治', 'で', '販売']
['カルビー', 'の', '約']
['立ち寄っ', 'て', 'ハロー']


 46%|████▌     | 637/1386 [01:14<00:45, 16.61it/s]

['系列', 'の', 'ずっと']
['系列', 'の', 'ずっと']
['プロポーズ', 'の', '年越し']


 46%|████▌     | 641/1386 [01:14<00:43, 17.16it/s]

['系列', 'の', 'ザ']
['妻', 'を', '一筋']
['番組', 'の', '御飯']
['喜劇', 'で', '放送']


 46%|████▋     | 644/1386 [01:15<00:42, 17.54it/s]

['喜劇', 'の', '明石家']
['本当', 'は', 'こんな']
['さんま', 'の', '侭']
['静岡', 'と', 'マドリード']
['キー', 'の', '抹茶']


 47%|████▋     | 647/1386 [01:15<00:37, 19.58it/s]

['味', 'は', 'You']
['静岡', 'の', '水曜']
['土産', 'の', 'あまおう']


 47%|████▋     | 652/1386 [01:15<00:46, 15.67it/s]

['社', 'か', 'ループ']
['。', 'の', '収録']
['楽曲', 'を', '著名']


 47%|████▋     | 656/1386 [01:15<00:51, 14.24it/s]

['など', 'が', 'You']
['ベースボール', 'に', '入団']
['投手', 'が', 'シンガポール']


 48%|████▊     | 661/1386 [01:16<00:45, 15.91it/s]

['シンガポール', 'の', '東北']
['東北', 'の', '松坂']


 48%|████▊     | 665/1386 [01:16<00:43, 16.44it/s]

['ライオンズ', 'の', 'カミング']
['沖縄', 'の', 'カミング']
['赤', 'の', '漆塗り']


 48%|████▊     | 669/1386 [01:16<00:47, 14.99it/s]

['安室', 'の', 'オリオン']
['で', 'は', '鸚哥']
['級', 'の', '水族']


 49%|████▊     | 673/1386 [01:16<00:49, 14.28it/s]

['リーズ', 'の', 'イエス']
['アギーレ', 'の', '誕生']
['代表', 'の', 'EU']


 49%|████▉     | 681/1386 [01:17<00:46, 15.27it/s]

['タイム', 'や', '冬']
['パスポート', 'の', '海豹']


 50%|████▉     | 687/1386 [01:17<00:35, 19.55it/s]

['園', 'の', 'バター']
['海豹', 'を', '間近']
['ドラマ', 'の', '黄色い']
['コロッケ', 'より', 'オランダ']


 50%|████▉     | 690/1386 [01:17<00:33, 20.81it/s]

['ゴッホ', 'や', 'ハビエル・アギーレ']
['アギーレ', 'や', 'フェルメール']
['味', 'の', 'ココ']


 50%|█████     | 693/1386 [01:18<01:03, 10.91it/s]

['情報', 'を', '御洒落']
['オランダ', 'で', '展示']


 50%|█████     | 697/1386 [01:18<00:54, 12.62it/s]

['テレビ', 'の', 'メロン']
['タイガース', 'の', 'BANDAI']
['ファイターズ', 'の', 'BANDAI']
['中止', 'で', '思い出し']


 51%|█████     | 700/1386 [01:18<00:50, 13.62it/s]

['パン', 'は', 'スーパーマーケット']
['皮', 'より', '重量']


 51%|█████     | 707/1386 [01:20<01:37,  6.95it/s]

['受け取り', 'が', '元気']
['ジャパン', 'の', 'ハンバーガー']
['アナ', 'と', '雪']
['州', 'と', 'スペイン']
['スペイン', 'や', 'マドリード']


 51%|█████▏    | 711/1386 [01:20<01:10,  9.64it/s]

['ストア', 'の', '湘南']
['ストア', 'の', '湘南']
['菓子', 'の', 'リプトン']
['0', 'の', 'スキー']


 52%|█████▏    | 717/1386 [01:20<00:45, 14.83it/s]

['選手', 'の', 'ドゥンガ']
['チュウ', 'の', 'レプリカ']
['未だ', 'に', 'アップル']
['八幡平', 'より', 'もう']


 52%|█████▏    | 723/1386 [01:21<00:39, 16.66it/s]

['土産', 'に', '島歌']
['ガーデン', 'の', '島歌']
['今年', 'の', 'グラッチェ']
['種類', 'だけ', 'メニュー']


 52%|█████▏    | 725/1386 [01:21<00:38, 17.16it/s]

['し', 'ながら', 'みけ']
['台湾', 'と', '東北']
['ソックス', 'の', '事故']


 53%|█████▎    | 730/1386 [01:21<00:40, 16.36it/s]

['時代', 'に', '御嶽']
['とんねるず', 'の', '皆']
['ノンストップ', 'も', '、']
['テレビ', 'の', '博士']


 53%|█████▎    | 736/1386 [01:21<00:31, 20.70it/s]

['菓子', 'の', 'リプトン']
['奈生子', 'の', '絵画']
['大阪', 'の', 'ミッドフィルダー']
['ストア', 'の', 'セレッソ']


 53%|█████▎    | 739/1386 [01:21<00:30, 20.90it/s]

['大阪', 'は', '今期']
['札幌', 'の', '長友']


 54%|█████▍    | 745/1386 [01:22<00:39, 16.37it/s]

['誠', 'が', 'コンサドーレ']
['。', 'の', '他']
['マート', 'の', 'リーダー']
['半島', 'の', 'You']
['←', 'の', 'You']


 54%|█████▍    | 750/1386 [01:22<00:39, 16.07it/s]

['キャンデー', 'で', '見']
['俳優', 'の', 'ヒカ']
['アーチスト', 'は', 'どんな']
['の', 'が', 'ツイッター']
['ツイッター', 'が', '漫画']


 55%|█████▍    | 756/1386 [01:22<00:32, 19.68it/s]

['フール', 'に', 'ツイッター']
['茅ヶ崎', 'と', 'アムステルダム']
['スターズ', 'の', '曲']
['バンド', 'の', 'サザン']
['FC', 'の', '楽曲']


 55%|█████▍    | 759/1386 [01:23<00:32, 19.23it/s]

['活躍', 'で', '茅ヶ崎']
['FC', 'の', '楽曲']
['ひろ子', 'と', '鉄拳']


 55%|█████▍    | 762/1386 [01:23<00:33, 18.71it/s]

['私', 'も', 'ブラタモリ']
['NHK', 'の', '町歩き']


 56%|█████▌    | 770/1386 [01:23<00:33, 18.12it/s]

['ピッチング', 'の', '安心']
['横綱', 'は', '海外']
['さび', 'で', '考え']
['映画', 'の', '宇多田']


 56%|█████▌    | 773/1386 [01:23<00:34, 17.69it/s]

['英', 'の', 'オーディション']
['タレント', 'の', '収録']
['台湾', 'の', '宇多田']
['サカナクション', 'の', '他']
['トキオ', 'の', 'ライオン']


 56%|█████▌    | 776/1386 [01:24<00:32, 19.04it/s]

['ボランチ', 'は', '栗原']
['料理', 'の', '牛蒡']


 56%|█████▋    | 782/1386 [01:24<00:31, 19.29it/s]

['大阪', 'の', 'あさイチ']
['サッカー', 'の', '川澄']
['選手', 'の', '御陰']
['大阪', 'の', 'アンデルソン']


 57%|█████▋    | 787/1386 [01:24<00:33, 17.80it/s]

['山形', 'も', '勢い']
['省', 'が', '総額']


 57%|█████▋    | 793/1386 [01:25<00:42, 13.99it/s]

['院', 'の', 'ソレデモシタイ']
['院', 'の', 'バラード']
['稼働', 'なんて', '、']


 57%|█████▋    | 795/1386 [01:25<00:40, 14.52it/s]

['朝日', 'の', '石井']
['朝日', 'の', '出入']
['系列', 'の', 'さんま']


 58%|█████▊    | 799/1386 [01:25<00:44, 13.27it/s]

['テニス', 'の', '矢部']
['浩之', 'は', '人気']
['ナイン', 'で', '放送']
['番組', 'の', 'ぶー']


 58%|█████▊    | 804/1386 [01:26<00:37, 15.37it/s]

['さん', 'の', 'ぶー']
['館', 'と', '国営']
['県', 'が', '桜ん坊']
['スーパーマーケット', 'の', 'や']


 58%|█████▊    | 807/1386 [01:26<00:32, 17.88it/s]

['フーズ', 'の', 'グループ']
['年', 'の', 'ソントン']


 59%|█████▊    | 811/1386 [01:26<00:35, 16.32it/s]

['楽曲', 'の', 'タイムズ']
['スマップ', 'の', '浜崎']
['さん', 'か', '宇多田']
['beta', 'に', '参加']


 59%|█████▉    | 820/1386 [01:27<00:55, 10.28it/s]

['楽曲', 'を', '著名']
['Tube', 'の', 'プロモーション']
['芸人', 'の', 'マツコ']
['芸人', 'の', 'マツコ']


 60%|█████▉    | 826/1386 [01:27<00:35, 15.68it/s]

['商品', 'を', 'マツコ']
['俳優', 'の', 'マツコ']
['マツコ', 'の', '知ら']
['俳優', 'の', 'マツコ']
['グリコ', 'の', '鸚哥']
['グリコ', 'の', 'グランド']


 60%|█████▉    | 829/1386 [01:27<00:33, 16.62it/s]

['グリコ', 'の', 'ぐり']
['グリコ', 'の', '竹の子']
['グリコ', 'の', '苺']
['ラインアップ', 'の', '中']


 60%|██████    | 835/1386 [01:28<00:28, 19.28it/s]

['ポッキー', 'が', '大阪']
['グリコ', 'の', '広島']
['園', 'と', '横川']


 60%|██████    | 838/1386 [01:28<00:30, 17.91it/s]

['ビール', 'の', 'ハッピー']
['ビール', 'の', 'スナック']
['自宅', 'で', 'アサヒ']
['ビール', 'の', 'スナック']
['ビール', 'の', 'かっぱ']


 61%|██████    | 844/1386 [01:28<00:28, 18.87it/s]

['選手', 'の', '三都主']
['レプリカ', 'が', '懸賞']
['コロッケ', 'は', '自動']
['日本', 'の', 'モンテディオ']


 61%|██████    | 848/1386 [01:28<00:29, 18.55it/s]

['コメ', 'を', 'ブラジル']
['。', 'が', '世界']
['タイガース', 'の', 'キャンデー']
['ホールディングス', 'の', '話題']
['ドーナツ', 'は', '量']


 62%|██████▏   | 854/1386 [01:29<00:26, 20.28it/s]

['中村', 'や', '香川']
['し', 'て', 'カミング']
['オランダ', 'も', '世界']
['沖縄', 'の', 'オランダ']


 62%|██████▏   | 857/1386 [01:29<00:25, 20.57it/s]

['オランダ', 'で', '展示']
['マート', 'の', '方々']
['ストア', 'の', 'スイーツ']
['声優', 'の', 'fukase']


 62%|██████▏   | 860/1386 [01:29<00:24, 21.64it/s]

['ロココ', 'を', '見る']


 62%|██████▏   | 865/1386 [01:29<00:33, 15.40it/s]

['院', 'が', '買う']


 63%|██████▎   | 868/1386 [01:30<00:29, 17.70it/s]

['letao', 'を', '応援']
['年始', 'に', 'ヨーロッパ']


 63%|██████▎   | 875/1386 [01:30<00:38, 13.24it/s]

['グランパス', 'の', 'フレーバー']
['女優', 'の', 'ジーコ']
['インド', 'で', 'ジーコ']
['イーグルス', 'の', '事故']


 63%|██████▎   | 878/1386 [01:30<00:33, 15.26it/s]

['東北', 'が', '桜ん坊']
['ガーデン', 'の', '事故']
['ガーデン', 'の', '鈴木']
['ガーデン', 'の', '鈴木']


 64%|██████▎   | 881/1386 [01:31<00:28, 17.52it/s]

['ガーデン', 'の', '鈴木']


 64%|██████▍   | 887/1386 [01:31<00:28, 17.73it/s]

['孝之', 'の', '東京']
['プロデューサー', 'に', 'テレビ']
['東京', 'の', 'もやもや']


 65%|██████▍   | 896/1386 [01:31<00:22, 21.66it/s]

['アナ', 'と', 'エルサ']
['逢', 'の', '話題']
['pasco', 'の', 'ファン']
['あなた', 'が', 'pasco']
['で', 'は', 'pasco']


 65%|██████▌   | 903/1386 [01:32<00:21, 22.71it/s]

['司会', 'の', '信長']
['。', 'の', '実写']
['スマップ', 'の', 'アナウンサー']
['スマップ', 'の', 'ファースト']


 66%|██████▌   | 910/1386 [01:32<00:18, 26.30it/s]

['神戸', 'と', '裏']
['IKEA', 'は', 'ござい']
['アンパンマンミュージアム', 'は', '神戸']
['IKEA', 'の', '鸚哥']
['IKEA', 'の', '鸚哥']
['アンパンマンミュージアム', 'は', '神戸']
['館', 'の', '鸚哥']


 66%|██████▌   | 916/1386 [01:32<00:18, 24.88it/s]

['神戸', 'の', '京都']
['館', 'の', '鸚哥']
['チャイ', 'は', '商品']
['チャイ', 'は', 'どちら']
['日本', 'と', '愛媛']


 67%|██████▋   | 922/1386 [01:32<00:19, 23.67it/s]

['日本', 'と', '四国']
['ハンバーガー', 'で', 'モ']
['年', 'も', 'MOS']
['リゾート', 'と', '伊勢丹']


 67%|██████▋   | 925/1386 [01:32<00:21, 21.31it/s]

['アナ', 'と', 'エルサ']
['吹き替え', 'や', '歌']
['0', 'の', '恋']
['グループ', 'の', 'akb']
['0', 'の', '軍師']


 67%|██████▋   | 931/1386 [01:33<00:21, 21.47it/s]

['0', 'の', '軍師']
['0', 'の', '国分']
['制作', 'の', '天空']
['選手', 'が', 'ルーム']


 68%|██████▊   | 937/1386 [01:33<00:19, 23.59it/s]

['ストア', 'の', 'ブンデスリーガ']
['今', 'や', '洋食']


 68%|██████▊   | 947/1386 [01:33<00:19, 22.55it/s]

['ランチ', 'も', '食べれる']


 69%|██████▊   | 950/1386 [01:40<04:54,  1.48it/s]

['意外', 'と', '少なく']
['と', 'か', 'スープ']


 70%|██████▉   | 965/1386 [01:41<01:04,  6.53it/s]

['か', 'も', '必須']
['です', 'ね', '一人']
['と', 'か', '天国']


 70%|███████   | 971/1386 [01:41<00:43,  9.47it/s]

['なん', 'か', 'サークル']


 71%|███████   | 978/1386 [01:41<00:29, 13.97it/s]

['バンク', 'の', 'ゼロ']


 71%|███████   | 984/1386 [01:42<00:23, 17.31it/s]

['なん', 'か', '響き']


 72%|███████▏  | 992/1386 [01:42<00:22, 17.18it/s]

['しよう', 'かしら', '。']
['深海', 'を', '操作']
['マニピュレーター', 'と', 'か']


 72%|███████▏  | 999/1386 [01:43<00:18, 20.51it/s]

['いう', 'か', '…']


 73%|███████▎  | 1007/1386 [01:43<00:14, 26.92it/s]

['柴犬', 'や', 'と']
['レトリーバー', 'と', '柴']
['レトリーバー', 'と', '柴']
['のばし', 'ちゃ', '駄目']


 74%|███████▎  | 1019/1386 [01:43<00:13, 26.39it/s]

['いう', 'か', '…']


 74%|███████▍  | 1028/1386 [01:44<00:15, 23.53it/s]

['編集', 'と', 'か']
['遣い', 'ながら', '吸い']
['です', 'か', '大晦日']


 75%|███████▍  | 1038/1386 [01:44<00:21, 16.12it/s]

['年始', 'も', '0']
['N', 'を', 'y']
['度', 'と', '競演']


 75%|███████▌  | 1044/1386 [01:45<00:19, 17.45it/s]

['夜行', 'も', '面白い']
['グッド', 'だけ', 'です']


 76%|███████▌  | 1054/1386 [01:45<00:14, 22.51it/s]

['因み', 'に', '昨日']


 77%|███████▋  | 1068/1386 [01:46<00:13, 24.33it/s]

['因み', 'に', 'あなた']
['です', 'かー', '。']
['いつ', 'も', '家']


 78%|███████▊  | 1076/1386 [01:46<00:10, 29.83it/s]

['暗く', 'まで', 'なり']
['一人', 'の', '時']


 78%|███████▊  | 1080/1386 [01:46<00:12, 23.65it/s]

['意外', 'と', '地味']
['と', 'か', '自然']
['定番', 'の', 'クライン']
['ガルボ', 'って', 'しっ']


 78%|███████▊  | 1083/1386 [01:46<00:12, 24.28it/s]

['プロ', 'に', 'なろう']
['と', 'か', '自然']
['大物', 'か', 'も']


 79%|███████▉  | 1092/1386 [01:47<00:13, 21.84it/s]

['て', 'から', '声掛け']
['互い', 'に', '見']
['ん', 'が', '…']
['か', 'も', '必須']


 79%|███████▉  | 1095/1386 [01:47<00:14, 19.67it/s]

['か', 'が', 'ひどかっ']


 80%|███████▉  | 1102/1386 [01:47<00:11, 24.20it/s]

['酒', 'は', '酎ハイ']
['酒', 'は', '酎ハイ']
['一人', 'の', '方']


 80%|███████▉  | 1108/1386 [01:47<00:11, 24.65it/s]

['指定', 'の', 'かっぱ']
['と', 'か', '表情']
['言葉', 'が', '切磋']
['一人', 'の', '方']
['御陰', 'で', '気づい']


 81%|████████  | 1124/1386 [01:48<00:09, 28.11it/s]

['因み', 'に', '足']
['同士', 'で', '忘年']


 82%|████████▏ | 1132/1386 [01:48<00:08, 28.52it/s]

['なん', 'か', 'サークル']
['手伝い', 'すら', 'ろくに']


 83%|████████▎ | 1144/1386 [01:49<00:07, 32.31it/s]

['ウインズ', 'って', 'あの']
['瀕し', 'て', '手術']


 83%|████████▎ | 1157/1386 [01:49<00:06, 33.30it/s]

['け', 'で', '掃除']
['机', 'なんて', '論外']
['と', 'か', '天国']


 85%|████████▍ | 1173/1386 [01:50<00:07, 28.99it/s]

['すみこ', 'の', 'メーク']


 85%|████████▌ | 1184/1386 [01:50<00:06, 30.50it/s]

['趣味', 'の', '白土']
['モード', 'の', 'なれ']
['機会', 'と', '気合い']


 86%|████████▋ | 1198/1386 [01:50<00:06, 27.66it/s]

['割', 'も', '夏']


 87%|████████▋ | 1205/1386 [01:51<00:07, 24.22it/s]

['だ', 'か', '取り留め']
['麺類', 'で', '済ま']


 88%|████████▊ | 1213/1386 [01:51<00:05, 29.37it/s]

['です', 'けど', '名店']
['ダンク', 'は', '好き']
['き', 'て', '院生']
['機', 'なんか', 'ダイソン']


 89%|████████▉ | 1236/1386 [01:52<00:10, 14.94it/s]

['対策', 'と', 'か']
['だ', 'と', 'ぼろ負け']


 89%|████████▉ | 1240/1386 [01:53<00:09, 15.53it/s]

['どう', 'か', '分かり']
['しまえ', 'ば', '規則']


 91%|█████████ | 1258/1386 [01:53<00:04, 27.84it/s]

['人類', 'に', '一人']
['いう', 'と', '一人']
['と', 'か', '外']
['飲め', 'て', '食べれ']


 92%|█████████▏| 1269/1386 [01:54<00:04, 27.33it/s]

['対し', 'て', '好き嫌い']


 92%|█████████▏| 1276/1386 [01:54<00:04, 27.50it/s]

['オー', 'と', 'か']


 93%|█████████▎| 1283/1386 [01:54<00:03, 28.04it/s]

['味醂', 'を', '0']
['料理', 'も', 'さっさと']
['に', 'は', '敵い']
['です', 'かー', '。']


 93%|█████████▎| 1287/1386 [01:54<00:03, 27.53it/s]

['です', 'か', '年始']
['ボール', 'は', '食え']
['雑誌', 'が', '0']


 94%|█████████▎| 1298/1386 [01:55<00:03, 27.70it/s]

['か', 'も', '必須']
['と', 'か', '日本']
['部活', 'から', 'ずっと']


 94%|█████████▍| 1304/1386 [01:55<00:02, 27.51it/s]

['練習', 'と', 'か']
['特番', 'で', '長かっ']
['因み', 'に', '何']
['鴨の嘴', 'と', 'いう']


 95%|█████████▍| 1310/1386 [01:55<00:02, 26.35it/s]

['かい', 'の', '時期']
['言葉', 'が', '切磋']


 95%|█████████▌| 1320/1386 [01:55<00:02, 28.06it/s]

['言葉', 'が', '切磋']
['た', 'から', '…']


 96%|█████████▌| 1327/1386 [01:56<00:02, 28.57it/s]

['時代', 'だけ', 'バス']
['虎', 'と', '黒猫']


 96%|█████████▌| 1333/1386 [01:56<00:02, 24.14it/s]

['ゼミ', 'に', '外国']


 97%|█████████▋| 1340/1386 [01:56<00:01, 27.15it/s]

['相乗り', 'と', 'か']


 97%|█████████▋| 1348/1386 [01:56<00:01, 30.72it/s]

['か', 'は', '作り置き']
['何', 'の', 'アンコール']


 98%|█████████▊| 1358/1386 [01:57<00:01, 23.88it/s]

['さん', 'で', '育た']
['デジカメ', 'は', 'もと']


 98%|█████████▊| 1364/1386 [01:57<00:00, 23.80it/s]

['送迎', 'に', '親']
['昼間', 'と', 'か']


 99%|█████████▉| 1376/1386 [01:58<00:00, 28.40it/s]

['因み', 'に', '足']
['因み', 'に', '足']
['それ', 'は', '仕方ない']
['作っ', 'て', 'いやいや']


100%|██████████| 1386/1386 [01:58<00:00, 11.68it/s]

['実', 'は', '。']


In [13]:
y_pred_2.count(0)

765

In [18]:
score(y, y_pred_2)

confusion matrix = 
 [[760 619]
 [  5   2]]
accuracy =  0.5497835497835498
precision =  0.00322061191626409
recall =  0.2857142857142857
f1 score =  0.006369426751592357


- 使い方が難しい

        confusion matrix = 
        [[666 713]
        [  6   1]]
        accuracy =  0.48124098124098125
        precision =  0.0014005602240896359
        recall =  0.14285714285714285
        f1 score =  0.0027739251040221915

- ちょっと改善(プログラムミスを修正)

        confusion matrix = 
        [[870 509]
        [  6   1]]
        accuracy =  0.6284271284271284
        precision =  0.00196078431372549
        recall =  0.14285714285714285
        f1 score =  0.0038684719535783366

    - 意外と普通っぽい ngram がない
        - ["を", "覚える", *]

    - スコア化はせず， ngram が存在しなければアウトと仮定した
        - 先行研究のモデルで，そういったボーダーは存在しない

- 用法エラー + 誤情報

        confusion matrix = 
        [[758 225]
        [118 285]]
        accuracy =  0.7525252525252525
        precision =  0.5588235294117647
        recall =  0.707196029776675
        f1 score =  0.6243154435925521
    

    

In [15]:
data_path = "../X_y_data/base_y_pred/"
# data_name = "utt_form.pickle"
data_name = "utt_content.pickle"
dataM = DataManager(data_path)
# dataM.save_data(data_name, [y, y_pred_2])

In [16]:
df = analyzer_unidic.analyze_with_dataframe("頼りがいのあった可愛くない兄さんが死んだ")
df

surface part_of_speech infl_type infl_form base_form  \
0     頼りがい    名詞-普通名詞-一般-                         タヨリガイ   
1        の       助詞-格助詞--                             ノ   
2       あっ     動詞-非自立可能--     五段-ラ行   連用形-促音便        アル   
3        た         助動詞---     助動詞-タ    連体形-一般         タ   
4      可愛く       形容詞-一般--       形容詞    連用形-一般      カワイイ   
5       ない    形容詞-非自立可能--       形容詞    連体形-一般        ナイ   
6        兄    名詞-普通名詞-一般-                            アニ   
7       さん    接尾辞-名詞的-一般-                            サン   
8        が       助詞-格助詞--                             ガ   
9       死ん        動詞-一般--     五段-ナ行   連用形-撥音便        シヌ   
10       だ         助動詞---     助動詞-タ    連体形-一般         タ   

                             reading phonetic  
0   [頼り甲斐, 頼りがい, タヨリガイ, 頼りがい, タヨリガイ]        *  
1                    [の, の, ノ, の, ノ]        *  
2               [有る, あっ, アッ, ある, アル]        *  
3                    [た, た, タ, た, タ]        *  
4        [可愛い, 可愛く, カワイク, 可愛い, カワイー]        *  
5               [無い, ない, ナイ, ない, ナイ]        *  
6                  [兄, 兄, アニ, 兄, アニ]        *  
7               [さん, さん, サン, さん, サン]        *  
8                    [が, が, ガ, が, ガ]        *  
9               [死ぬ, 死ん, シン, 死ぬ, シヌ]        *  
10                   [た, だ, ダ, だ, ダ]        *